cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/BoardsConfigService.kt

```diff
@@ -12,8 +13,13 @@ class BoardsConfigService(
     private val storage: Storage = boards(),
     private val contentRepository: ContentRepository = ContentRepository()
 ) {
+    private var boardDeletionHelper: BoardDeletionHelper? = null
+
     fun saveBoardsConfig(boardsConfig: JsonObject): Boolean {
         val clearBoardsConfig = executeForWidgets(boardsConfig, this::clearContent)
+
+        handleDeletedBoards(boardsConfig)
```

Comment:
IMO: This is a good place to remove unwanted widget content

Reply:
Fixed somewhat, see newest commits.